homeassistant/components/livisi/coordinator.py

```diff
@@ -122,6 +122,16 @@ async def async_get_vrcc_humidity(self, capability: str) -> int | None:
             return None
         return response["humidity"]["value"]
 
+    async def async_get_wds_state(self, capability: str) -> bool | None:
+        """Get the window door sensor state."""
+        response: dict[str, Any] = await self.aiolivisi.async_get_device_state(
+            capability[1:]
+        )
+        if response is None:
+            return None
+        on_state = response["isOpen"]
+        return on_state["value"]
+
```

Comment:
There's a lot of duplicate code here. Would something like this work?
```python
    async def _async_get_device_response(self, capability, key):
        response: dict[str, Any] = await self.aiolivisi.async_get_device_state(
            capability[1:]
        )
        if response is None:
            return None
        return response[key]["value"]

    async def async_get_wds_state(self, capability: str) -> bool | None:
        """Get the window door sensor state."""
        return await self._async_get_device_response(capability, "isOpen")
```

And maybe `async_get_wds_state` should be in the `binary_sensor` module if it is binary-sensor specific.

Reply:
You are right. In my view the coordinator just needs `async_get_device_response` and the rest should be handled in the devices. I just implemented it this way because switch and climate do it the same way... I will update the code, but for consistency climate and switch should also be updated.